webapp/src/main/java/life/qbic/usermanagement/registration/RegistrationEmailSender.java

```diff
@@ -0,0 +1,75 @@
+package life.qbic.usermanagement.registration;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.Properties;
+import javax.mail.Authenticator;
+import javax.mail.Message;
+import javax.mail.Message.RecipientType;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * Sends emails informing a user that she was registered in the system
+ *
+ * @since 1.0.0
+ */
+@Service
+public class RegistrationEmailSender {
```

Comment:
Instead of having a solution that cannot be re-used, I favour an EmailSubmissionService, that provides some public methods, without knowing anything about the email purpose. The service needs a recipient email address and a pre-formatted content. 

We can then re-use the service, for example when we confirm the registration and activate the user account. 

This implementation cannot be re-used for other emails unfortunately.

I suggest to wrap the the registration email logic in a EmailComposer factory, that creates Email objects. An Email has a recipient and a message content + subject. 

You can for example then have a call like:

`Email registrationMail = EmailComposer.createRegistrationMail(String recipientEmail, String subject, String content)`

And pass it to the EmailSubmissionService.

Reply:
I think that makes the solution way too complex. Generalisation makes sense when we need the functionality in other spaces. Currently, we only need it in this function.

Reply:
But why is it complex? It fits into the current application architecture and separates the concerns. Your suggestion puts formatting and submission into one class. 

Also, it does not fit in the Service Layout we have.